### PR TITLE
SCB-2256 The address of the config center of the service startup log is printed automatically according to the component type

### PR DESCRIPTION
**Pull request by develpoerX**
…is printed automatically according to the component type

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by develpoerX** [comment]
I have test this modify. it works well. And I have modify some old codes to rebuild them.

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.Arrays;
+
+public class ConfigCenterDefaultDeploymentProvider implements DeploymentProvider {
+  public static final String SYSTEM_KEY_CONFIG_CENTER = "ConfigCenter";
+
+  private static AbstractConfiguration configuration = ConfigUtil.createLocalConfig();
+
+  @Override
+  public SystemBootstrapInfo getSystemBootStrapInfo(String systemKey) {
+    if (systemKey.contentEquals(SYSTEM_KEY_CONFIG_CENTER)) {
```
Comment (liubao68) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java:35]: use SYSTEM_KEY_CONFIG_CENTER.equals(systemKey)

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java [hunk 2]
```diff
@@ -15,44 +15,37 @@
...
-        String[] urls = configuration.getStringArray("servicecomb.service.registry.address");
-        if (urls == null || urls.length == 0) {
-          urls = new String[] {"http://127.0.0.1:30100"};
-        }
-        sc.setAccessURL(Arrays.asList(urls));
-        return sc;
-      case SYSTEM_KEY_CONFIG_CENTER:
-        String[] ccAddresses = configuration.getStringArray("servicecomb.config.client.serverUri");
-        if (ccAddresses == null || ccAddresses.length == 0) {
+    if (systemKey.contentEquals(SYSTEM_KEY_KIE_CENTER)) {
```
Comment (liubao68) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java:35]: same as above

File: huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java [hunk 3]
```diff
@@ -31,23 +31,21 @@
...
-      case MonitorConstant.SYSTEM_KEY_DASHBOARD_SERVICE:
-        String[] msAddresses = configuration.getStringArray(MonitorConstant.MONITOR_URI);
-        if (msAddresses == null || msAddresses.length == 0) {
-          return null;
-        }
-        SystemBootstrapInfo ms = new SystemBootstrapInfo();
-        ms.setAccessURL(Arrays.asList(msAddresses));
-        return ms;
-      default:
+    if (systemKey.contentEquals(MonitorConstant.SYSTEM_KEY_DASHBOARD_SERVICE)) {
```
Comment (liubao68) [inline huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java:39]: same as above

File: deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java [hunk 4]
```diff
@@ -17,23 +17,32 @@
...
-    switch (systemKey) {
-      case DeploymentProvider.SYSTEM_KEY_CONFIG_CENTER:
-        SystemBootstrapInfo cc = new SystemBootstrapInfo();
-        cc.setAccessURL(Arrays.asList("http://lcalhost/custom"));
-        return cc;
-      default:
-        return null;
+    if (systemKey.contentEquals("TestCenter")) {
+      SystemBootstrapInfo cc = new SystemBootstrapInfo();
+      cc.setAccessURL(Arrays.asList("http://lcalhost/custom"));
```
Comment (AngLi2) [inline deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java:38]: typo

File: deployment/src/test/java/org/apache/servicecomb/deployment/TestDeployment.java [hunk 5]
```diff
@@ -31,45 +31,9 @@ public void tearDown() {
 
   @Test
   public void testConfiguration() {
-    DefaultDeploymentProvider.setConfiguration(ConfigUtil.createLocalConfig());
-    SystemBootstrapInfo info = Deployment.getSystemBootStrapInfo(DeploymentProvider.SYSTEM_KEY_SERVICE_CENTER);
-    Assert.assertEquals(info.getAccessURL().get(0), "http://127.0.0.1:30100");
-    info = Deployment.getSystemBootStrapInfo(DeploymentProvider.SYSTEM_KEY_CONFIG_CENTER);
+    CustomDeploymentProvider.setConfiguration(ConfigUtil.createLocalConfig());
+    SystemBootstrapInfo info = Deployment.getSystemBootStrapInfo("TestCenter");
     Assert.assertEquals(info.getAccessURL().get(0), "http://lcalhost/custom");
```
Comment (AngLi2) [inline deployment/src/test/java/org/apache/servicecomb/deployment/TestDeployment.java:36]: typo "lcalhost"

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/collect/ServiceCenterDefaultDeploymentProvider.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  public static final String SYSTEM_KEY_SERVICE_CENTER = "ServiceCenter";
+
+  private static AbstractConfiguration configuration = ConfigUtil.createLocalConfig();
+
+  @Override
+  public SystemBootstrapInfo getSystemBootStrapInfo(String systemKey) {
+    if (systemKey.contentEquals(SYSTEM_KEY_SERVICE_CENTER)) {
+      SystemBootstrapInfo sc = new SystemBootstrapInfo();
+      String[] urls = configuration.getStringArray("servicecomb.service.registry.address");
+      if (urls == null || urls.length == 0) {
```
Comment (AngLi2) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/collect/ServiceCenterDefaultDeploymentProvider.java:38]: use ArrayUtils.isEmpty(urls)

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      String[] ccAddresses = configuration.getStringArray("servicecomb.config.client.serverUri");
+      if (ccAddresses == null || ccAddresses.length == 0) {
```
Comment (AngLi2) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java:37]: use ArrayUtils.isEmpty(ccAddresses)

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java [hunk 8]
```diff
@@ -15,44 +15,37 @@
...
+      String[] kieAddresses = configuration.getStringArray("servicecomb.kie.serverUri");
+      if (kieAddresses == null || kieAddresses.length == 0) {
```
Comment (AngLi2) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java:37]: use ArrayUtils.isEmpty(kieAddresses)

File: huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java [hunk 9]
```diff
@@ -31,23 +31,21 @@
...
+      String[] msAddresses = configuration.getStringArray(MonitorConstant.MONITOR_URI);
+      if (msAddresses == null || msAddresses.length == 0) {
```
Comment (AngLi2) [inline huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java:41]: use ArrayUtils.isEmpty(msAddresses)

File: deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java [hunk 10]
```diff
@@ -17,23 +17,32 @@
...
+      return cc;
     }
+    return null;
```
Comment (AngLi2) [inline deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java:41]:
```java
if (!systemKey.contentEquals("TestCenter")) {
  return null;
}
xxx
xxx
```
seems to be better

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/archaius/sources/ConfigCenterConfigurationSourceImpl.java [hunk 11]
```diff
@@ -65,7 +64,7 @@ public int getOrder() {
 
   @Override
   public boolean isValidSource(Configuration localConfiguration) {
-    if (Deployment.getSystemBootStrapInfo(DeploymentProvider.SYSTEM_KEY_CONFIG_CENTER) == null) {
+    if (Deployment.getSystemBootStrapInfo(ConfigCenterDefaultDeploymentProvider.SYSTEM_KEY_CONFIG_CENTER) == null) {
```
Comment (AngLi2) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/archaius/sources/ConfigCenterConfigurationSourceImpl.java:67]: return Deployment.getSystemBootStrapInfo(ConfigCenterDefaultDeploymentProvider.SYSTEM_KEY_CONFIG_CENTER) != null

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java [hunk 12]
```diff
@@ -102,18 +102,18 @@ public String getToken() {
...
-	} else if (property instanceof List) {
-	  return (List<String>) property;
-	}
-	return new ArrayList<>();
+    Object property = finalConfig.getProperty(FILE_SOURCE);
+    if (property instanceof String) {
+      List<String> result = new ArrayList<>();
+      result.add((String) property);
+      return result;
+    } else if (property instanceof List) {
```
Comment (AngLi2) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java:113]: "else" is redundant after "return"

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java [hunk 13]
```diff
@@ -102,18 +102,18 @@ public String getToken() {
...
+      return (List<String>) property;
+    }
+    return new ArrayList<>();
```
Comment (AngLi2) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java:116]: `return Collections.EMPTY_LIST` is better

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        return null;
+      }
+      SystemBootstrapInfo cc = new SystemBootstrapInfo();
+      cc.setAccessURL(Arrays.asList(ccAddresses));
+      return cc;
+    }
+    return null;
```
Comment (AngLi2) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java:44]:
```java
if (aaa){
  return null;
}
xxx;
xxx;
xxx;
```
seems to be better

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterInformationCollector.java [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+    return "Config Center: " + getCenterInfo(Deployment.getSystemBootStrapInfo("ConfigCenter"));
+  }
+
+  @Override
+  public int getOrder() {
+    return 1;
+  }
+
+  private String getCenterInfo(SystemBootstrapInfo systemBootstrapInfo) {
+    if (systemBootstrapInfo == null) {
```
Comment (AngLi2) [inline dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterInformationCollector.java:37]: `return Objects.isNull(systemBootstrapInfo)?"not exist":systemBootstrapInfo.getAccessURL().toString();`

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java [hunk 16]
```diff
@@ -15,44 +15,37 @@
...
-        cc.setAccessURL(Arrays.asList(ccAddresses));
-        return cc;
-      default:
-        return null;
+      }
+      SystemBootstrapInfo kie = new SystemBootstrapInfo();
+      kie.setAccessURL(Arrays.asList(kieAddresses));
+      return kie;
     }
+    return null;
```
Comment (AngLi2) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java:44]: same as above

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieClientInformationCollector.java [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+  @Override
+  public int getOrder() {
+    return 2;
+  }
+
+  private String getCenterInfo(SystemBootstrapInfo systemBootstrapInfo) {
+    if (systemBootstrapInfo == null) {
+      return ("not exist");
+    }
+    return systemBootstrapInfo.getAccessURL().toString();
```
Comment (AngLi2) [inline dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieClientInformationCollector.java:40]: same as above

use ternary operator

File: huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java [hunk 18]
```diff
@@ -31,23 +31,21 @@
...
         return null;
+      }
+      SystemBootstrapInfo ms = new SystemBootstrapInfo();
+      ms.setAccessURL(Arrays.asList(msAddresses));
+      return ms;
     }
+    return null;
```
Comment (AngLi2) [inline huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java:48]: same as above

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/collect/ServiceCenterDefaultDeploymentProvider.java [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+        urls = new String[]{"http://127.0.0.1:30100"};
+      }
+      sc.setAccessURL(Arrays.asList(urls));
+      return sc;
+    }
+    return null;
```
Comment (AngLi2) [inline service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/collect/ServiceCenterDefaultDeploymentProvider.java:44]: same as above

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterInformationCollector.java (hunk 15 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterInformationCollector.java:37]: Does this compound statement comply with the clean code specification?

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java (hunk 14 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java:44]: ok。

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java (hunk 1 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java:35]: done

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java (hunk 2 above)
Comment (develpoerX) [reply dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java:35]: done

File: huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java (hunk 3 above)
Comment (develpoerX) [reply huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java:39]: done

File: deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java (hunk 4 above)
Comment (develpoerX) [reply deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java:38]: done

File: deployment/src/test/java/org/apache/servicecomb/deployment/TestDeployment.java (hunk 5 above)
Comment (develpoerX) [reply deployment/src/test/java/org/apache/servicecomb/deployment/TestDeployment.java:36]: done

File: huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java (hunk 9 above)
Comment (develpoerX) [reply huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java:41]: done

File: deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java (hunk 10 above)
Comment (develpoerX) [reply deployment/src/test/java/org/apache/servicecomb/deployment/CustomDeploymentProvider.java:41]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java (hunk 14 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterDefaultDeploymentProvider.java:44]: done

File: service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/collect/ServiceCenterDefaultDeploymentProvider.java (hunk 19 above)
Comment (develpoerX) [reply service-registry/registry-service-center/src/main/java/org/apache/servicecomb/serviceregistry/collect/ServiceCenterDefaultDeploymentProvider.java:44]: done

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java (hunk 16 above)
Comment (develpoerX) [reply dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieCenterDefaultDeploymentProvider.java:44]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterInformationCollector.java (hunk 15 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/collect/ConfigCenterInformationCollector.java:37]: done

File: huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java (hunk 18 above)
Comment (develpoerX) [reply huawei-cloud/dashboard/src/main/java/org/apache/servicecomb/huaweicloud/dashboard/monitor/MonitorDefaultDeploymentProvider.java:48]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/archaius/sources/ConfigCenterConfigurationSourceImpl.java (hunk 11 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/archaius/sources/ConfigCenterConfigurationSourceImpl.java:67]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java (hunk 13 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java:116]: done

File: dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java (hunk 12 above)
Comment (develpoerX) [reply dynamic-config/config-cc/src/main/java/org/apache/servicecomb/config/client/ConfigCenterConfig.java:113]: done

File: dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieClientInformationCollector.java (hunk 17 above)
Comment (develpoerX) [reply dynamic-config/config-kie/src/main/java/org/apache/servicecomb/config/kie/collect/KieClientInformationCollector.java:40]: done

**Review by liubao68 — APPROVED** [review]

**Pull request merged**